src/config-service.ts

```diff
@@ -21,14 +22,78 @@ export interface IConfigService {
   ): Promise<ISyncOptions>
 }
 
-const explorer = cosmiconfig('typesync')
+async function loadYaml(_filepath: string, content: string): Promise<unknown> {
+  return await yaml.parse(content)
+}
+
+// From
+function searchPlaces(moduleName: string) {
+  const rcs = [
+    `.${moduleName}rc`,
+    `.${moduleName}rc.json`,
+    `.${moduleName}rc.yaml`,
+    `.${moduleName}rc.yml`,
+    `.${moduleName}rc.js`,
+    `.${moduleName}rc.ts`,
+    `.${moduleName}rc.mjs`,
+    `.${moduleName}rc.cjs`,
+  ]
+
+  return [
+    'package.json',
+    ...rcs,
+    ...rcs.map((rc) => `.config/${rc}`),
+    `${moduleName}.config.js`,
+    `${moduleName}.config.ts`,
+    `${moduleName}.config.mjs`,
+    `${moduleName}.config.cjs`,
+  ]
+}
+
+async function loadTs(filename: string): Promise<unknown> {
+  if (process.features.typescript) {
+    return await loadJs(filename)
+  }
+
+  if (import.meta.resolve('jiti')) {
+    const jiti = (await import('jiti')).createJiti(import.meta.url)
+
+    return await jiti.import(filename, { default: true })
+  }
```

Comment:
Happy to drop this if we don't care about (undocumented) TS support.

Reply:
Meh, can keep it I guess. Can make it clear in the error that to load a ts config file, install jiti. However, most people likely use typesync via npx - does npx install peer deps?

Reply:
It does, but not _optional_ peer deps.

Reply:
I think so